### DISCUSSION TITLE
Conditional config based on the project I'm in

### DISCUSSION THREAD

**Discussion by ryanwinchester** [vyfor/cord.nvim#138] [Q&A] [answered]
I want to hide/change some things if I'm working on work stuff vs my own stuff.

I'm new to neovim, so not sure if this is just a normal and easy thing to do or not and have no idea how to do it.

**Comment by vyfor** [comment]
Hey!
Currently, the only way to hide the rich presence is by either using the `display.workspace_blacklist` setting in the config for specific workspace names or by setting up a new autocmd to manually invoke `:CordShowPresence` and `:CordHidePresence` based on specific conditions. If you need more flexibility, I’m working on a project rewrite that will make customization incredibly versatile and cover every use case. If you are interested and able to share more details about your use case (e.g., what you’d like to change and under what exact conditions), I’ll be happy to provide more guidance.

**Comment by ryanwinchester** [comment]
```lua
local cwd = vim.fn.fnamemodify(vim.fn.getcwd(), ":t")
local is_work = (cwd == "acme")

return {
  "vyfor/cord.nvim",
  build = "./build || .\\build",
  event = "VeryLazy",
  opts = { -- calls require('cord').setup()
    display = {
      show_time = true, -- Display start timestamp
      show_repository = not is_work, -- Display 'View repository' button linked to repository url, if any
      show_cursor_position = false, -- Display line and column number of cursor's position
      swap_fields = false, -- If enabled, workspace is displayed first
      swap_icons = false, -- If enabled, editor is displayed on the main image
      workspace_blacklist = {}, -- List of workspace names that will hide rich presence
    },
    text = {
      viewing = is_work and "Viewing [redacted]" or "Viewing {}",
      editing = is_work and "Editing [redacted]" or "Editing {}",
      file_browser = is_work and "Browsing [redacted]" or "Browsing files in {}",
      plugin_manager = is_work and "Managing plugins at work" or "Managing plugins in {}",
      lsp_manager = is_work and "Configuring LSP at work" or "Configuring LSP in {}",
      vcs = is_work and "Committing work" or "Committing changes in {}",
      workspace = is_work and "At work" or "In {}",
    },
  },
}
```

this seems to be working

![image](https://github.com/user-attachments/assets/18d6e669-172b-487b-a3d9-f209c4972453)

![image](https://github.com/user-attachments/assets/ac78070d-57ee-4e7d-971d-4ddd0e51e981)

but might have issues

[![Screenshot 2024-12-09@14 47 57 @2x](https://github.com/user-attachments/assets/b6818d4a-2555-476e-8a40-c3e3178f9bd2)](https://x.com/teej_dv/status/1866180836418359489)

**Reply by vyfor** [reply, marked as answer]
That might work, but it won’t be dynamic. It will persist for the entire session based on where Neovim is opened. You'd need to restart Neovim each time you want to change the workspace. But unfortunately, there's no way around this, which is why a rewrite is being done. I'm currently in the process of writing the new configuration guide, would you like me to notify you when it's ready?

**Reply by ryanwinchester** [reply]
yeah, that would be great
